script/color-contrast.ts

```diff
@@ -0,0 +1,173 @@
+ * @description test the contrast of two colors against each other
+ * @param minimumContrast 
+ * @param colorA 
+ * @param colorB 
+ * @param bg used to calculate an opaque color if colorB is semi transparent
+ * @param contrastPair use for better error messages
+ * @returns 
+ */
+const testContrast = (minimumContrast: number, colorA: string, colorB: string, bg: string = '#ffffff', contrastPair?: string): { pass: string, contrastRatio: string } => {
+  // get contrast
+  let contrast = 0
+  try {
+    colorB = getOpaqueColor(colorB, bg)
+    colorA = getOpaqueColor(colorA, colorB)
+    // get contrast rounded down with 2 decimals
+    contrast = Math.floor(getContrast(colorA, colorB) * 100) / 100
+  } catch (err) {
+    console.error(`${contrastPair || ""} as ${colorA} vs.${colorB}: ${err}`)
+  }
+  return {
+    pass: contrast >= minimumContrast ? '✅' : '❌',
+    contrastRatio: `${contrast}: 1`
+  }
+}
+/**
+ * renderConsoleTable
+ * @description takes the test results per theme and prints a nicely formatted table of the results to the console 
+ * @param theme 
+ * @param results 
+ */
+const renderConsoleTable = (theme: string, results: contrastTestResult[]): void => {
+  // config table
+  const contrastTable = new Table({
+    title: `Contrast checks for: ${theme}`,
+    charLength: { "❌": 2, "✅": 2 },
+    colorMap: {
+      grey: '\x1b[0;30m', // define customized color
+    },
+    columns: [
+      {
+        name: "contrastPair",
+        alignment: "left",
+        title: "Color pair",
+      },
+      {
+        name: "pass",
+        alignment: "center",
+        title: "Pass",
+      },
+      {
+        name: "contrastRatio",
+        alignment: "left",
+        title: "Contrast ratio"
+      },
+      {
+        name: "minimumContrastRatio",
+        alignment: "left",
+        title: "Min. ratio"
+      }]
+  })
+  // add rows and color
+  for (const [index, row] of results.entries()) {
+    let color = "white"
+    // turn odd index rows grey (index starts at 0)
+    if (index % 2 !== 0) {
+      color = "grey"
+    }
+    contrastTable.addRow(row, {
+      color: color
+    })
+  }
+  // print table to console
+  console.log(contrastTable.render())
+}
+/**
+ * run tests, output results to console and store them for json
+ */
+const results = Object.entries(contrastRequirements)
+  .map(([theme, colorPairs]: [theme: string, colorPairs: ContrastRequirement[]]) => {
+    // turn deeply nested colors object into one level object like 'fg.default': '#000'
+    // @ts-ignore
+    const flattenColors = flattenObject(colors.colors[theme])
+    // run tests on all color pairs
+    const results = runContrastTest(colorPairs, flattenColors)
+    // print results to console
+    renderConsoleTable(theme, results)
+    // return results for json file creation
+    return {
+      theme,
+      results
+    }
+  })
+
+// write json file for workflow
+fs.writeFile('color-contrast-check.json', JSON.stringify(results), (err) => {
```

Comment:
Can you put the results into the `dist/` folder please? That's on a gitignore so it won't create any unstaged changes on the CI server when/if we do auto-commits from there. I'll still be able to use it for workflows from there. 

```suggestion
const outputDirectory = '/tokens-v2-private/json';
if (!fs.existsSync(outputDirectory)){
    fs.mkdirSync(outputDirectory);
}
fs.writeFile(`${outputDirectory}/color-contrast-check.json`, JSON.stringify(results), (err) => {
```

Reply:
Done.